### DISCUSSION TITLE
Weird/different behaviour of `obj.downcast::<PyIterator>()` on Windows vs. MacOS and Linux

### DISCUSSION THREAD

**Discussion by samuelcolvin** [PyO3/pyo3#2902] [Questions] [answered]
See https://github.com/pydantic/pydantic-core/pull/367 and the associated failing test.

Effectively, `obj.downcast::<PyIterator>().is_ok()` is `true` on windows, but `false` on macos and linux, for the following input.

```py
from collections import Sequence

class MySequence(Sequence):
    def __init__(self):
        self._data = [1, 2, 3]

    def __getitem__(self, index):
        return self._data[index]

    def __len__(self):
        return len(self._data)

input_value = MySequence()
```

Anyone have any idea why this is? Could this be a bug in pyo3 or cpython?

@davidhewitt any input would be great.

**Comment by davidhewitt** [comment, marked as answer]
For the record, follow up in #2913
